pkg/image/resolve.go

```diff
@@ -119,21 +119,32 @@ func generateImages(chartNameAndVersion string, inputMap map[interface{}]interfa
 		return
 	}
 
-	// distinguish images by os
-	os := inputMap["os"]
-	switch os {
-	case "windows": // must have indicate `os: windows` if the image is using in Windows cluster
-		if osType != Windows {
-			return
+	imageName := fmt.Sprintf("%s:%v", repo, t)
+
+	// By default, images are added to the generic images list ("linux"). For Windows and multi-OS
+	// images to be considered, they must use a comma-delineated list (e.g. "os: windows",
+	// "os: windows,linux", and "os: linux,windows").
+	if osList, ok := inputMap["os"].(string); ok {
+		for _, os := range strings.Split(osList, ",") {
+			switch strings.TrimSpace(strings.ToLower(os)) {
+			case "windows":
+				if osType == Windows {
+					addSourceToImage(output, imageName, chartNameAndVersion)
```

Comment:
Don't you need return here, so it doesn't get added multiple times ? Also need to squash commits

Reply:
No, because the osType will be the same throughout the loop. Returning is optional